.github/workflows/compile_engine.yml

```diff
@@ -0,0 +1,53 @@
+name: Release binaries for circus-engine
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  release:
+    if: startsWith(github.ref, 'refs/tags/')
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          # - windows-latest
+        ocaml-compiler: [4.14.x]
+        
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - uses: actions-rs/toolchain@v1
+        id: rust_toolchain
+        with:
+          toolchain: stable
+          
+      - run: cargo install --path cli
```

Comment:
Why do we need the cli when building the engine?

Reply:
The Rust part (aka the cli) gives the types to the ML part, it provides a binary `cargo-circus-export-json-schema` (or something) that outputs JSON schema, which is then processed by a script (the dirty JS one) called by dune to create a ML module of input types